data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/configuration/S3SelectJsonOptionTest.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.configuration;
+
+import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.test.helper.ReflectivelySetField;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class S3SelectJsonOptionTest {
+
+    @Test
+    void s3SelectJsonOptionWithLinesTest() throws NoSuchFieldException, IllegalAccessException {
+        S3SelectJsonOption jsonOption = new S3SelectJsonOption();
+        ReflectivelySetField.setField(S3SelectJsonOption.class,jsonOption,"type","Lines");
```

Comment:
Rather that set the field, parse from a YAML string. See my comment above.

Reply:
Thanks David for your review, we have incorporated the changes.